inspire_utils/name.py

```diff
@@ -246,6 +248,104 @@ def from_parts(
+             but not from ``Smith, Jane``.
+            3) ``a J Smith``
+            This will just generate a ``match`` query on ``last_name`` and a match query on ``first_name.initials``.
+            Please note, cases such as ``J.D.`` have been properly handled by the tokenizer.
+        """
+        nested_query = {
+                'nested': {
+                    'path': keyword,
+                    'query': {
+                        'bool': {
+                            'must': [],
+                        },
+                    },
+                },
+            }
+
+        def _match_query_with_names_initials_analyzer_with_and_operator(field, value):
+            return {
+                "match": {
+                    field: {
+                        "query": value,
+                        'operator': 'AND',
+                        "analyzer": "names_initials_analyzer"
+                    }
+                }
+            }
+
+        def _match_query_with_and_operator(field, value):
+            return {
+                'match': {
+                    field: {
+                        'query': value,
+                        'operator': 'AND'
+                    }
+                }
+            }
+
+        def _match_phrase_prefix_query(field, value):
+            return {
+                "match_phrase_prefix": {
+                    field: {
+                        "query": value,
+                        "analyzer": "names_analyzer"
+                    }
+                }
+            }
+
+        if len(self) == 1 and '.' not in self.first:
+            # ParsedName returns first name if there is only one name i.e. `Smith`
+            # in our case we consider it as a lastname
+            last_name = self.first
+            query = _match_query_with_and_operator('{}.last_name'.format(keyword), last_name)
+            nested_query['nested']['query']['bool']['must'].append(query)
+            return nested_query
+
+        bool_query_build = [_match_query_with_and_operator('{}.last_name'.format(keyword), self.last)]
+        author_names = [re.split(r'\.(?=[A-Za-z]|\s|$)', name) for name in self.first_list]
+        first_names = filter(None, chain.from_iterable(author_names))
+
+        should_query = []
+        for name in first_names:
+            name_query = []
+            if len(name) == 1 or '.' in name:
+                name_query.append(
+                    _match_query_with_names_initials_analyzer_with_and_operator(
+                        '{}.first_name.initials'.format(keyword), name)
+                )
+            else:
+                name_query.extend([
+                    _match_phrase_prefix_query('{}.first_name'.format(keyword), name),
+                    _match_query_with_names_initials_analyzer_with_and_operator('{}.first_name'.format(keyword), name)
+                ])
+            if '.' not in self.first and ',' not in author_name:
```

Comment:
Ok, so this is what you need the `author_name` for. This is used in the query parser to determine because there's an ambiguity: is `a John Smith` a search for `first="John", second="Smith"` or `second="John Smith"`? The proper way to have this information is to do this check in the `__init__`, and set a flag such as `maybe_only_last_name` that you can then test here.